### ISSUE TITLE
Deploy a secondary worker process from a single repo

### ISSUE THREAD

**Issue by enricodeleo** [githubsaturn/captainduckduck#251]
Maybe it is related to #37 but as a Heroku, Dokku and Flynn user I'm used to the possibility of spawning a different process from a unique app/repository. This is useful for example if you use a queue worker that is based on the very same codebase of the web-facing application. 

I was wondering if it is possible somehow with captainduckduck or I have to deploy the same app twice with same env variables just because on the second one I have a different entrypoint (thus I have to remember to double-deploy the same app everytime I update the codebase).

**Comment by githubsaturn** [comment]
Procfile terminology is specific to Heroku. Docker, however, uses a generic ENTRYPOINT and CMD command. I'll try to briefly explain how to achieve what you are looking for:
- Case 1) You want to run two processes. Perhaps a webserver, and another process which polls data from an RSS feed. In this case, look at [this example](https://docs.docker.com/config/containers/multi-service_container/). Your first process is can be your webserver, your second process can be your RSS feed crawler.
- Case 2) you want to run a one-off process with each deploy. This can be a database migration script. The best approach is to use the case that is described as Case 1, but your script needs to have an internal check where it checks the database has already been updated or not. The reason is that this script will always run with each container restart. So you want to make sure that your script can handle if it runs multiple times.
- Case 3) Similar to case 1, but you only want to run your RSS reader every 5 hours. Do exactly what case 1 explains. Except, instead of process 2 being a continuous process, you should just setup a CRON job.

Note that in the example that I linked above, if any of the processes exits, the container restarts. In case 2 and case 3, you only care about process 1. Process 2 is a temporary process in case 2 and case 3. Therefore, in these cases, the script needs to be adjusted. 

In a nutshell, with some simple bash scripting, you can spin off a secondary process if you want.


PS:
If you want to run a script after each DEPLOY or CONFIGURATION UPDATE, take a look at [post-deploy hook](https://captainduckduck.com/docs/pre-deploy-script.html).
